python/src/schema.rs

```diff
@@ -1008,14 +1008,69 @@ impl PySchema {
     /// Return equivalent PyArrow schema
     ///
     /// :rtype: pyarrow.Schema
-    #[pyo3(text_signature = "($self)")]
-    fn to_pyarrow(self_: PyRef<'_, Self>) -> PyResult<PyArrowType<ArrowSchema>> {
+    #[pyo3(text_signature = "($self, as_large_types)")]
+    fn to_pyarrow(
+        self_: PyRef<'_, Self>,
+        as_large_types: Option<bool>,
+    ) -> PyResult<PyArrowType<ArrowSchema>> {
         let super_ = self_.as_ref();
-        Ok(PyArrowType(
-            (&super_.inner_type.clone())
-                .try_into()
-                .map_err(|err: ArrowError| PyException::new_err(err.to_string()))?,
-        ))
+        let res: ArrowSchema = (&super_.inner_type.clone())
+            .try_into()
+            .map_err(|err: ArrowError| PyException::new_err(err.to_string()))?;
+
+        fn convert_to_large_type(field: ArrowField, dt: ArrowDataType) -> ArrowField {
+            match dt {
+                ArrowDataType::Utf8 => field.with_data_type(ArrowDataType::LargeUtf8),
+
+                ArrowDataType::Binary | ArrowDataType::FixedSizeBinary(_) => {
+                    field.with_data_type(ArrowDataType::LargeBinary)
+                }
+
+                ArrowDataType::List(f) | ArrowDataType::FixedSizeList(f, _) => {
+                    let sub_field = convert_to_large_type(*f.clone(), f.data_type().clone());
+                    field.with_data_type(ArrowDataType::LargeList(Box::from(sub_field)))
+                }
```

Comment:
Why convert fixed size list to large list? It doesn't have any indices, so it doesn't have a small / large variant. Is this just because Polars doesn't support FSL?

Reply:
It doesn't seem like this is necessary to make the PR work, so I'd propose we change this to keep FSB and FSL as the same type. We still should transform the inner type of FSL like we do other nested types.

Reply:
yes you are correct, this was mostly for compatibility, but we can also do the cast or error handling for unsupported types in Polars directly.

Made the requested changes.